cmake/nanobind-config.cmake

```diff
@@ -4,20 +4,40 @@ if (NOT TARGET Python::Module)
   message(FATAL_ERROR "You must invoke 'find_package(Python COMPONENTS Interpreter Development REQUIRED)' prior to including nanobind.")
 endif()
 
-# Determine the Python extension suffix and stash in the CMake cache
-execute_process(
-  COMMAND "${Python_EXECUTABLE}" "-c"
-    "import sysconfig; print(sysconfig.get_config_var('EXT_SUFFIX'))"
-  RESULT_VARIABLE NB_SUFFIX_RET
-  OUTPUT_VARIABLE NB_SUFFIX
-  OUTPUT_STRIP_TRAILING_WHITESPACE)
-
-if (NB_SUFFIX_RET AND NOT NB_SUFFIX_RET EQUAL 0)
-  message(FATAL_ERROR "nanobind: Python sysconfig query to "
-    "find 'EXT_SUFFIX' property failed!")
+# Determine the right suffix for ordinary and stable ABI extensions
+if (Python_SOSABI)
+  set(NB_SUFFIX_EXT "${CMAKE_SHARED_MODULE_SUFFIX}")
+  if (WIN32)
+    set(NB_SUFFIX_EXT ".pyd")
+  endif()
+
+  # These are guaranteed to both be available in CMake 3.26+
+  set(NB_SUFFIX   ".${Python_SOABI}${NB_SUFFIX_EXT}")
+  set(NB_SUFFIX_S ".${Python_SOSABI}${NB_SUFFIX_EXT}")
```

Comment:
Wouldn't this have two `..` if `Python_SOSABI` is empty?

Reply:
yes, it would.. 🤦